api/src/main/java/org/apache/iceberg/view/SQLViewRepresentation.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.util.List;
+import org.apache.iceberg.catalog.Namespace;
+
+public interface SQLViewRepresentation extends ViewRepresentation {
+
+  @Override
+  default Type type() {
+    return Type.SQL;
+  }
+
+  /** The view query SQL text. */
+  String query();
+
+  /** The view query SQL dialect. */
+  String dialect();
+
+  /** The default catalog when the view is created. */
+  String defaultCatalog();
+
+  /** The default namespace when the view is created. */
+  Namespace defaultNamespace();
+
+  /**
+   * The output column names of the query that creates this view. The field aliases are not applied.
+   */
+  List<String> queryColumnNames();
```

Comment:
Why is this not a schema?

Reply:
Thinking about this more, I think it would make sense if this were a schema ID rather than query column names. Since this is derived from the query by the engine that creates the view, we should keep the full version.

We may also want to simply remove this and add a schema ID to `ViewVersion` since the schema should be consistent across representations.

Reply:
Also relevant discussion here: https://github.com/apache/iceberg/pull/6134#discussion_r1015129057

I think we should try to avoid thinking in terms what Spark stores in its metadata and try to understand the higher level objective.

Reply:
I think "original" query column names are specifically useful for `SELECT *` queries only, and for full utility they should be accompanied by table names as well to avoid ambiguity. I think we should also decouple the data communicated from an engine to the catalog as part of a `CREATE VIEW` statement, and the what we capture in the View Representation. Having `ViewRepresentation` heavily influenced by `CREATE VIEW` syntax could be the reason for this complexity?

Adding the right tag for Erik @xkrogen.

Reply:
Yeah, lots of complexity comes from

- column alias feature of CREATE VIEW in Spark
- store column names for SELECT star query

Reply:
@jzhuge, I think that we need the schema stored and accessible, and I think that should be the pre-alias schema that is derived from the SQL query.

Storing the pre-alias column names isn't sufficient because engines may choose to validate more than just the top-level columns that are selected. For example, an engine may want to validate that the type of an `id` column is still a bigint and not, for example, a string. The same logic also applies to nested types for evolution.

The most strict validation of a view is to ensure that the SQL currently produces the same schema, or one that can be cast to the same schema.

Reply:
I updated this. Will merge when tests are passing.